bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,248 @@
+spec:
+  azureName: #@ data.values.resource_group.name
+  location: #@ data.values.location
+#@ tags_list = []
+#@ tags_list.extend(data.values.global_tags)
+#@ tags_list.extend(data.values.resource_group.tags)
+#@ tags = {x["key"]: x["value"] for x in tags_list}
+#@ if/end len(tags) > 0:
+  tags: #@ tags
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersDatabase
+metadata:
+  name: #@ data.values.name
+  namespace: #@ data.values.namespace
+spec:
+  azureName: #@ data.values.database.name
+  owner:
+    name: #@ flexibleServerName
+  charset: utf8
+#@ tags_list = []
+#@ tags_list.extend(data.values.global_tags)
+#@ tags_list.extend(data.values.database.tags)
+#@ tags = {x["key"]: x["value"] for x in tags_list}
+#@ if/end len(tags) > 0:
+  tags: #@ tags
+
+#@ fwRuleCounter = 0
+#@ for rule in data.values.firewall_rules:
+#@   startAddr = ip.parse_addr(rule.startIpAddress)
+#@   endAddr = ip.parse_addr(rule.endIpAddress)
+#@   tags_list = []
+#@   tags_list.extend(data.values.global_tags)
+#@   tags_list.extend(rule.tags)
+#@   tags = {x["key"]: x["value"] for x in tags_list}
+--- 
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersFirewallRule
+metadata:
+  name: #@ data.values.name + "-" + str(fwRuleCounter)
+  namespace: #@ data.values.namespace
+spec:
+  owner:
+    name: #@ flexibleServerName
+  startIpAddress: #@ startAddr.string()
+  endIpAddress: #@ endAddr.string()
+#@   if/end len(tags) > 0:
+  tags: #@ tags
+#@   fwRuleCounter += 1
+#@ end  
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServer
+metadata:
+  name: #@ flexibleServerName
+  namespace: #@ data.values.namespace
+spec:
+  location: #@ data.values.location
+  azureName: #@ flexibleServerName
+  owner:
+    name: #@ resourceGroupK8SName
+  version: #@ data.values.server.version
+  sku:
+    name: #@ data.values.server.instance_type
+    tier: #@ data.values.server.instance_tier
+  administratorLogin: #@ data.values.server.administrator_name
+  administratorLoginPassword:
+    name: #@ dbSecretName
+    key: password
+  storage:
+    storageSizeGB: #@ data.values.server.instance_storage_size_gb
+#@ tags_list = []
+#@ tags_list.extend(data.values.global_tags)
+#@ tags_list.extend(data.values.server.tags)
+#@ tags = {x["key"]: x["value"] for x in tags_list}
+#@ if/end len(tags) > 0:
+  tags: #@ tags
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.namespace
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.name
+- apiGroups:
+  - dbforpostgresql.azure.com
+  resources:
+  - flexibleservers
+  - flexibleserversdatabases
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.name
```

Comment:
_question_: It seems you force the user to specify a database name, but here you default to the server's name. Isn't that a mismatch? Won't this only work when the database name is the same as the server name?

Reply:
See: https://github.com/vmware-tanzu/tanzu-application-platform-reference-service-packages/pull/22/files#r975346168